dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventImporterValidationRunner.java

```diff
@@ -27,54 +27,63 @@
  */
 package org.hisp.dhis.dxf2.events.importer;
 
-import static org.apache.commons.logging.LogFactory.getLog;
+import static org.hisp.dhis.dxf2.importsummary.ImportStatus.ERROR;
+import static org.hisp.dhis.dxf2.importsummary.ImportStatus.WARNING;
 
+import java.util.ArrayList;
 import java.util.List;
 
-import org.apache.commons.logging.Log;
+import lombok.RequiredArgsConstructor;
+
 import org.hisp.dhis.dxf2.events.event.Event;
 import org.hisp.dhis.dxf2.events.importer.context.WorkContext;
+import org.hisp.dhis.dxf2.events.importer.shared.ImmutableEvent;
+import org.hisp.dhis.dxf2.importsummary.ImportSummary;
+import org.springframework.stereotype.Component;
 
-/**
- * Simple interface that provides processing capabilities on events.
- *
- * @author maikel arabori
- */
-public interface EventProcessing
+@Component
+public class EventImporterValidationRunner
 {
-    void process( WorkContext workContext, List<Event> events );
-
-    Log log = getLog( EventProcessing.class );
-
-    class ProcessorRunner
+    /**
+     * Validates the events using the supplied list of validators.
+     * <p>
+     * Only returns the ImportSummary for Events that *did* not pass validation
+     *
+     * @param payload Validation Runner payload to run against
+     * @return returns the ImportSummary for Events that did not pass validation
+     */
+    public List<ImportSummary> run( ValidationRunnerPayload payload )
     {
-        private final WorkContext workContext;
-
-        private final List<Event> events;
-
-        public ProcessorRunner( WorkContext workContext, List<Event> events )
-        {
-            this.workContext = workContext;
-            this.events = events;
-        }
+        final List<ImportSummary> importSummaries = new ArrayList<>( 0 );
 
-        public void run( final List<Class<? extends Processor>> processors )
+        for ( final Event event : payload.events )
         {
-            for ( final Event event : events )
+            for ( Checker validationCheck : payload.validators )
             {
-                for ( Class<? extends Processor> processor : processors )
+                final ImportSummary importSummary = validationCheck.check( new ImmutableEvent( event ),
+                    payload.workContext );
+
+                if ( importSummary.isStatus( ERROR ) || importSummary.isStatus( WARNING ) )
                 {
-                    try
-                    {
-                        final Processor pre = processor.newInstance();
-                        pre.process( event, workContext );
-                    }
-                    catch ( InstantiationException | IllegalAccessException e )
+                    importSummaries.add( importSummary );
+                    if ( importSummary.isStatus( ERROR ) )
                     {
-                        log.error( "An error occurred during Event import processing", e );
+                        break;
```

Comment:
This is one of these `break`s where I would wonder where exactly it would take me and which would take some time analysing the code more careful. Could we get away with a control flow construct that is more clear like returning or inverting a if condition?

Reply:
I understand your point, I don't like breaks as well, but unless introducing flags and navigating collections by index (which is even worst in my opinion), this can't be removed in an elegant way (if you have one please suggest). So in this case I'll live with break.
If we used at least java 9 API, we could use "takeWhileInclusive" of java stream, but I think our minimum API level is 8 (correct me if I'm wrong)

Reply:
It is indeed 11 since a few days :+1: 
But such a change is maybe better done in further PRs.